A more detailed description of methods.

Agalma v2.0.0 ([Dunn et al. 2023](http://dx.doi.org/10.1186/1471-2105-14-330), [Guang et al. 2021](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0244202)) was used to run QC, clean rRNA, and assemble a Trinity transcriptome [Grabherr et al. 2011](https://www.nature.com/articles/nbt.1883). Bowtie2 mapping rates were used to assess the assembly quality. [Transdecoder](https://github.com/TransDecoder/TransDecoder/wiki) was used for translation, and [OrthoFinder2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1832-y) was used to identify orthologous genes. Finally, trinotate [Bryant et al. 2022](https://www.cell.com/cell-reports/fulltext/S2211-1247(16)31770-3) was used to annotate transcripts and genes.